vnext/Microsoft.ReactNative/Fabric/Composition/SwitchComponentView.cpp

```diff
@@ -203,9 +203,24 @@ void SwitchComponentView::Draw() noexcept {
       thumbX = (trackMarginX + trackWidth - thumbRadius - thumbRadius - thumbMargin) * m_layoutMetrics.pointScaleFactor;
     }
 
-    m_thumbVisual.Size(
-        {thumbRadius * m_layoutMetrics.pointScaleFactor, thumbRadius * m_layoutMetrics.pointScaleFactor});
-    m_thumbVisual.Position({thumbX, thumbY});
+    // handles various mouse events
+    if (m_pressed && !switchProps->disabled) {
+      m_thumbVisual.AnimatePosition({thumbX - 0.8f, thumbY - 0.8f});
+    } else if (m_hovered && !switchProps->disabled) {
+      m_thumbVisual.Size(
+          {thumbRadius * m_layoutMetrics.pointScaleFactor + 0.8f,
+           thumbRadius * m_layoutMetrics.pointScaleFactor + 0.8f});
+      m_thumbVisual.Position({m_thumbVisual.Position().x - 0.8f, m_thumbVisual.Position().y - 0.8f});
```

Comment:
You shouldn't use the current position as an input here.  Otherwise, if the draw function is run multiple times, you'd end up having the position go negative.

Just use thumbX - 0.8f